### PR TITLE
`Whitelisting` settings finalization

### PR DESCRIPTION
**Pull request by mudrila**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -119,7 +119,7 @@
     "jsdom": "^24.0.0",
     "netlify-cli": "^17.33.3",
     "prettier": "^3.3.3",
-    "typescript": "^5.5.4",
+    "typescript": "^5.5.3",
```
Comment (mudrila) [inline package.json:122]: Was getting a very weird issues after moving types around and deleting `types/utils` - a mix of "moving type to different file" and downgrading TS version just 1 patch lower did the trick - not sure what da heck is going on here

File: src/components/Roles/RolesDetailsDrawer.tsx [hunk 2]
```diff
@@ -146,7 +147,7 @@ export default function RolesDetailsDrawer({
                 variant="tertiary"
                 size="icon-sm"
                 aria-label="Edit Role"
-                as={PencilLine}
+                as={PencilWithLineIcon}
```
Comment (mudrila) [inline src/components/Roles/RolesDetailsDrawer.tsx:150]: Noticed we were using wrong icon here and there so updated that as well

File: src/components/SafeSettings/SettingsNavigation.tsx [hunk 3]
```diff
@@ -36,7 +35,7 @@ function SettingsLink({
         isCurrentPath ||
         (!isMobile &&
           isIndexSettingsPage &&
-          pathWithoutSearch === `/${DAO_ROUTES.settingsGeneral.path}`)
+          pathWithoutSearch === `/${DAO_ROUTES.settings.path}/${DAO_ROUTES.settingsGeneral.path}`)
```
Comment (mudrila) [inline src/components/SafeSettings/SettingsNavigation.tsx:38]: @adamgall The change you've requested previously - have had to update it into string literal to keep consistency with router declaration

File: src/components/ui/modals/ConfirmDeleteStrategyModal.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        gap={2}
+      >
+        <Button
+          variant="primary"
+          onClick={onClose}
+        >
+          {t('nevermind')}
+        </Button>
+        <Button
+          variant="secondary"
```
Comment (mudrila) [inline src/components/ui/modals/ConfirmDeleteStrategyModal.tsx:32]: Handling clicking on this buddy comes in the next PR - I'll introduce some interesting approach to properly handle getting into proposal builder page without crazy conditional rendering on the settings screen. Bet we'd want to migrate Roles into this approach as well

File: src/constants/routes.ts [hunk 5]
```diff
@@ -85,22 +85,27 @@ export const DAO_ROUTES = {
...
   settingsGovernance: {
     relative: (addressPrefix: string, daoAddress: string) =>
       `/settings/governance${getDaoQueryParam(addressPrefix, daoAddress)}`,
-    path: 'settings/governance',
+    path: 'governance',
   },
   settingsModulesAndGuard: {
     relative: (addressPrefix: string, daoAddress: string) =>
       `/settings/modules-and-guard${getDaoQueryParam(addressPrefix, daoAddress)}`,
-    path: 'settings/modules-and-guard',
```
Comment (mudrila) [inline src/constants/routes.ts:98]: So this `settings/blah` thingy was causing a problem for routing definition so I've decided to revert this change

File: src/pages/dao/roles/SafeRolesPage.tsx [hunk 6]
```diff
@@ -103,37 +102,6 @@ export function SafeRolesPage() {
           </Show>
         </>
       )}
-
-      <Show below="md">
```
Comment (mudrila) [inline src/pages/dao/roles/SafeRolesPage.tsx:107]: Removed as per request from @parkermccurley

File: src/pages/dao/roles/edit/details/SafeRoleEditDetailsPage.tsx [hunk 7]
```diff
@@ -218,9 +218,7 @@ export function SafeRoleEditDetailsPage() {
           <Hide below="md">
             <ModalBase
               isOpen
-              title=""
```
Comment (mudrila) [inline src/pages/dao/roles/edit/details/SafeRoleEditDetailsPage.tsx:221]: Optional - gone 😊

File: src/types/fractal.ts [hunk 8]
```diff
@@ -149,12 +149,6 @@ export interface ITokenAccount {
   votingWeightString: string | undefined;
 }
 
-/**
- * @dev This interface represents the store for the Fractal DAO.
- * @param baseContracts - This object contains the base contracts for the Fractal DAO.
- * @param clients - This object contains the clients for the Fractal DAO.
```
Comment (mudrila) [inline src/types/fractal.ts:155]: Wtf is even `clients` lol

File: package.json (hunk 1 above)
Comment (mudrila) [reply package.json:122]: Now surpisingly when I've tried to move types from `types/utils` to `types/common` - that didn't work lol
Most likely - circle import problem in action 🤣

File: src/components/ui/page/Header/NestedPageHeader.tsx [hunk 9]
```diff
@@ -8,32 +8,34 @@ export default function NestedPageHeader({
...
+          color="neutral-7"
+          padding={0}
+          leftIcon={<ArrowLeft />}
+          {...buttonProps}
+        >
+          {backButtonText}
+        </Button>
+      </Box>
+      <Text textStyle="body-base-strong">{title}</Text>
+      {children || <Box width="25%" />}
```
Comment (Da-Colon) [inline src/components/ui/page/Header/NestedPageHeader.tsx:38]: Don't hate on the `??` :D

File: src/pages/dao/settings/permissions/SafePermissionsSettingsPage.tsx [hunk 10]
```diff
@@ -1,53 +1,94 @@
...
+                color="neutral-6"
+                onClick={() =>
+                  navigate(
+                    DAO_ROUTES.settingsPermissionsCreateProposal.relative(
+                      addressPrefix,
+                      safe.address,
+                    ),
+                  )
+                }
+              />
```
Comment (Da-Colon) [inline src/pages/dao/settings/permissions/SafePermissionsSettingsPage.tsx:74]: I could be missing it, but is this missing an `icon`?

File: src/components/SafeSettings/SettingsPermissionsStrategyForm.tsx [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+      </Flex>
+      <LabelWrapper
+        label={t('permissionAmountLabel')}
+        labelColor="neutral-7"
+      >
+        <BigIntInput
+          onChange={setProposerThreshold}
+          decimalPlaces={votesToken.decimals}
+          value={proposerThreshold.bigintValue}
+        />
```
Comment (Da-Colon) [inline src/components/SafeSettings/SettingsPermissionsStrategyForm.tsx:89]: I assume validation and such will all happen later?

**Review by Da-Colon — APPROVED** [review]
Nothing Merge-blocking. LGTM

File: src/components/ui/icons/PencilWithLineIcon.tsx
[the remark this answers is not in the document]
Comment (mudrila) [reply src/components/ui/icons/PencilWithLineIcon.tsx:1]: Oh, good one - I'll move this icon there

File: src/components/ui/page/Header/NestedPageHeader.tsx (hunk 9 above)
Comment (mudrila) [reply src/components/ui/page/Header/NestedPageHeader.tsx:38]: Oh it's not hate actually - `??` was causing `Box` not being rendered when `children` is `false`

File: src/pages/dao/settings/permissions/SafePermissionsSettingsPage.tsx (hunk 10 above)
Comment (mudrila) [reply src/pages/dao/settings/permissions/SafePermissionsSettingsPage.tsx:74]: You're absolutely right - icon is missing, thank you sir!

File: src/components/SafeSettings/SettingsPermissionsStrategyForm.tsx (hunk 11 above)
Comment (mudrila) [reply src/components/SafeSettings/SettingsPermissionsStrategyForm.tsx:89]: Yessir - I'll put all the actual logic and error handling into next PR

File: package-lock.json [hunk 12]
```diff
@@ -94,7 +94,7 @@
         "jsdom": "^24.0.0",
         "netlify-cli": "^17.33.3",
         "prettier": "^3.3.3",
-        "typescript": "^5.5.4",
+        "typescript": "^5.5.3",
```
Comment (adamgall) [inline package-lock.json:97]: @mudrila your "downgrade" didn't actually do anything. Notice how this line is the _only_ change in `package-lock.json`.

Because version `5.5.4` (what it was) satisfies the pattern `^5.5.3`, npm didn't actually downgrade the `typescript` package.

File: src/assets/theme/custom/icons/PencilWithLineIcon.tsx
Comment (adamgall) [inline src/assets/theme/custom/icons/PencilWithLineIcon.tsx:1]: I guess I'm confused. Why can't we use the default `PencilLine` icon exported from `@phosphor-icons/react`?

File: src/components/ui/page/Header/NestedPageHeader.tsx (hunk 9 above)
Comment (adamgall) [reply src/components/ui/page/Header/NestedPageHeader.tsx:38]: What is that Box anyway?

File: src/components/ui/page/Header/NestedPageHeader.tsx [hunk 13]
```diff
@@ -8,32 +8,34 @@ export default function NestedPageHeader({
...
   children,
-}: PropsWithChildren<{ title: string; backButtonText: string; backButtonHref: string }>) {
+  onGoBack,
+}: PropsWithChildren<{
+  title: string;
+  backButtonText: string;
+  backButtonHref: string;
+  onGoBack?: () => void;
+}>) {
+  const buttonProps = onGoBack ? { onClick: onGoBack } : { as: Link, to: backButtonHref };
```
Comment (adamgall) [inline src/components/ui/page/Header/NestedPageHeader.tsx:18]: I have an alternative solution to _yet another optional param_. please see my PR targeting this branch.

Comment (adamgall) [reply src/components/ui/page/Header/NestedPageHeader.tsx:18]: https://github.com/decentdao/decent-interface/pull/2452

File: src/components/ui/page/Header/NestedPageHeader.tsx (hunk 9 above)
Comment (mudrila) [reply src/components/ui/page/Header/NestedPageHeader.tsx:66]: Just a spacer that puts title in a proper place when `children` is missing

File: src/assets/theme/custom/icons/PencilWithLineIcon.tsx
Comment (mudrila) [reply src/assets/theme/custom/icons/PencilWithLineIcon.tsx:1]: Because it doesn't match the design (and this one looks better from my "personal user standpoint"

File: src/components/ui/page/Header/NestedPageHeader.tsx (hunk 13 above)
Comment (mudrila) [reply src/components/ui/page/Header/NestedPageHeader.tsx]: Yeah, I like your updates sir

**Review by adamgall — APPROVED** [review]
